packages/ui/src/components/Typography/Typography.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { ReactNode, ReactText } from 'react';
+import { Box, forwardRef, HTMLChakraProps } from '@chakra-ui/react';
+
+type TypographyVariant = 'span' | 'div' | 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+
+type TypographyFontSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl' | '5xl' | '6xl' | '7xl';
+
+type TypographyFontWeight = 'regular' | 'medium';
+
+type RemoveCommonValues<Type, TOmit> = {
+  [Property in keyof Type]: TOmit extends Record<Property, infer U> ? Exclude<Type[Property], U> : Type[Property];
+};
+
+type Id<Type> = Record<string, unknown> & { [P in keyof Type]: Type[P] };
+type ConditionalProps<Type, TKey extends keyof TCase, TCase extends Partial<Type>> =
+  | Id<Omit<Type, keyof TCase> & TCase>
+  | Id<RemoveCommonValues<Type, Pick<TCase, TKey>>>;
```

Comment:
I would recommend  to put RemoveCommonValues and ConditionalProps in other file (in shared forlder maybe?), because if anybody else will need that, it would be easy to use it then